ime/app/src/main/java/com/anysoftkeyboard/AnySoftKeyboard.java

```diff
@@ -662,6 +673,55 @@ private void onNonFunctionKey(
             default:
                 if (isWordSeparator(primaryCode)) {
                     handleSeparator(primaryCode);
+                    // retrieves the first two characters before cursor
+                    String previousTwoChars = ic.getTextBeforeCursor(2, 0).toString();
+                    String stringPrimaryCode = new String(new int[] {primaryCode}, 0, 1);
+                    boolean anyPuntuaction =
```

Comment:
Can we use `com.anysoftkeyboard.ime.AnySoftKeyboardSuggestions#isWordSeparator` instead?

Reply:
as I commented above, as of now, I can only apply those changes after `handleSeparator` is been called. Before or inside `handleSeparator` didn't work for me. I need to undesrstand how to do that first.

Reply:
sorry i didn't get your question right. 

`isWordSeparator` tells me if it is a separator or not. I need to know which separator is exactely. e.g `(` and `)` will be treated differently so I'm afraid isWordSeparator can't give me that information, as it is just a boolean.